editor/editor_data.cpp

```diff
@@ -509,6 +511,104 @@ EditorPlugin *EditorData::get_extension_editor_plugin(const StringName &p_class_
 	return plugin == nullptr ? nullptr : *plugin;
 }
 
+void EditorData::add_context_menu_plugin(ContextMenuSlot slot, const Ref<EditorContextMenuPlugin> &p_plugin) {
+	ContextMenu cm;
+	cm.p_slot = slot;
+	cm.plugin = p_plugin;
+	p_plugin->start_idx = context_menu_plugins.size() * EditorContextMenuPlugin::MAX_ITEMS;
+	context_menu_plugins.push_back(cm);
+}
+
+void EditorData::remove_context_menu_plugin(ContextMenuSlot slot, const Ref<EditorContextMenuPlugin> &p_plugin) {
+	for (int i = context_menu_plugins.size() - 1; i > -1; i--) {
+		if (context_menu_plugins[i].p_slot == slot && context_menu_plugins[i].plugin == p_plugin) {
+			context_menu_plugins.remove_at(i);
+		}
+	}
+}
+
+void EditorData::add_options_from_plugins(PopupMenu *p_popup, ContextMenuSlot p_slot, const Vector<String> &p_paths) {
+	bool add_separator = false;
+
+	for (int i = 0; i < context_menu_plugins.size(); i++) {
+		const ContextMenu &cm = context_menu_plugins[i];
+		if (cm.p_slot != p_slot) {
+			continue;
+		}
+		if (!add_separator) {
+			add_separator = true;
+			p_popup->add_separator();
```

Comment:
This should only happen after the condition below. Otherwise the menu can get an empty separator.
![image](https://github.com/user-attachments/assets/26e7ad45-dea8-4b71-8c49-697487f36aee)